api/app/Repositories/Criteria/SearchByNameStringCriteria.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Repositories\Criteria;
+
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Support\Facades\DB;
+
+final class SearchByNameStringCriteria implements Criteria
+{
+    private ?string $searchString;
+
+    public function __construct(?string $searchString)
```

Comment:
The `name` is required parameter. Do not create the criteria if the search string is not provided.